### PR TITLE
feat: Scenes and GameEntities

### PR DESCRIPTION
**Pull request by jksevend**
- Created an abstract `Scene` class 
- Created an abstract `GameEntity : RenderObject` class (Modified `RenderObject` to provide virtual `update` and `render` functions)
     - Maybe a bit confusing at the beginning but plan is to add a separate abstract `WidgetEntity : RenderObject` in the future, also considering the ECS can only be applied to  a `GameEntity` i think this makes sense
- `AssetManager` loads a Surface for now, normally i would create a texture here right away but we dont have access to it in the `Scene` constructor (Maybe let asset manager return a Todot defined texture) 

Have a look at the example, to see what my vision for the scenes and game entities are. There is a lot to improve im sure.

Also currently not implemented but maybe following tomorrow:

- Switch between scenes
- Background Color of scenes
- Destroying/Caching of resources when a scene is destroyed/disposed

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -32,6 +32,10 @@
         "xlocmon": "cpp",
         "xlocnum": "cpp",
         "xloctime": "cpp",
-        "xmemory": "cpp"
+        "xmemory": "cpp",
```
Comment (FrederikTobner) [inline .vscode/settings.json:35]: We should really delete this stupid file and add it to the .gitignore file in another pr😂

**Review by FrederikTobner — APPROVED** [review]
lgtm

File: .vscode/settings.json (hunk 1 above)
Comment (jksevend) [reply .vscode/settings.json:35]: Yes i really dont know why there are always changes. Although the conventionalCommits.scopes are also saved in here...

**Comment by FrederikTobner** [comment]
maybe we should add the build check in the branch protection rules before we allow merging to master.

**Comment by jksevend** [comment]
Dont know how to do that, lets have a look tomorrow

**Pull request merged**